src/main/scala/edu/berkeley/ce/sparkrocks/SeedJointSelector.scala

```diff
@@ -3,242 +3,221 @@ package edu.berkeley.ce.sparkrocks
       }
-      val nonRedundantBlocks = blocks map { case block @ Block(blockCenter, _, _) =>
-          Block(blockCenter, block.nonRedundantFaces)
+      // Insufficient partitions found
+      if (partitioning.length < numPartitions) {
+        findSeedBlocks(jointSets.tail, inputVolume, numPartitions, jointSetSpan, partitioning,
+          remainingJoints ++ jointSets.head)
+        // Sufficient partitions, return
+      } else {
+        (partitioning, remainingJoints ++ leftOverJoints)
       }
-      val volumes = nonRedundantBlocks.map(_.volume)
-      val avgVolume = volumes.sum / volumes.length
-      println("\nLOAD BALANCE INFORMATION:")
-      println(s"Average Volume: %.2f".format(avgVolume))
-      println(s"Max Volume: %.2f".format(volumes.max))
-      println(s"Min Volume: %.2f".format(volumes.min))
-      Some(seedJoints)
+      // Joint set non-persistent, check next one
+    } else {
+      findSeedBlocks(jointSets.tail, inputVolume, numPartitions, jointSetSpan, seedBlocks, remainingJoints)
     }
   }
 
   /**
-    * Iterates through input joint set and selects joints that divide the initial block into approximately
-    * equal volumes
+    * Search input joint sets for joints that divide input volume into approximately equal volumes
+    * based on desired number of partitions.
+    *
+    * @param jointSets Seq of Seq's of Joints representing input joint sets
+    * @param inputVolume Block representing input rock volume
+    * @param numPartitions Number of partitions desired
+    * @param jointSetSpan Number of joint sets to search at a time for partition generation
+    * @param spanCount Number or joint sets to search for seed joints
+    * @param seedJoints Seq of joints that were not used in seed block generation
+    * @param remainingJoints Seq of joints that were not selected seed joints
+    * @return Returns a tuple containing two Seq's of Joints. The first Seq contains the seed joints that
+    *         best divide the input volume. The second Seq contains the joints that remain from the joint
+    *         sets that have been checked.
+    */
+  @tailrec
+  def searchJointSets(jointSets: Seq[Seq[Joint]], inputVolume: Block, numPartitions: Int, jointSetSpan: Int,
+                      spanCount: Int = 1, seedJoints: Seq[Joint] = Seq.empty[Joint],
+                      remainingJoints: Seq[Joint] = Seq.empty[Joint]): (Seq[Joint], Seq[Joint]) = {
+    // Checked specified span of joint sets, return
+    if (spanCount > jointSetSpan) {
+      (seedJoints, remainingJoints)
+    } else {
+      val totalVolume = inputVolume.volume
+      // Determine remaining partitions that need to be found.
+      val remainingPartitions = numPartitions / jointSetSpan
+      val volumePerPiece = totalVolume / remainingPartitions
+      val newSeedJoints = findSeedJoints(jointSets.head, inputVolume, totalVolume, volumePerPiece)
+      val remainingFromJointSet = jointSets.head.diff(newSeedJoints)
```

Comment:
To avoid this `diff`, why not just have `findSeedJoints` return a pair with the first element as the chosen joints and the second as the unused joints?